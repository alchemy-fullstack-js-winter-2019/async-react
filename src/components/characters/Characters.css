.Characters {
    border: 3px solid green;
}