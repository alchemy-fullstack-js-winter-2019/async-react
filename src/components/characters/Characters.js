import React, { Component, Fragment } from 'react';
import Character from './Character';
import { getCharacters } from '../services/rickAndMortyApi';
import PropTypes from 'prop-types';
import styles from './Characters.css';


export default class App extends Component {
    // constructor(props) {
    //     super(props);
    // }
    static propTypes = {
        currentPage: PropTypes.number.isRequired,
        updateTotalPages: PropTypes.func.isRequired
    };
    state = {
        characters: []
    };

    componentDidUpdate() {
        getCharacters(this.props.currentPage)
            .then(chars => {
                this.setState({ characters: chars.results }, () => {
                    this.props.updateTotalPages(chars.totalPages);
                });
            });

    }
    componentDidMount() {
        getCharacters(this.props.currentPage)
            .then(chars => {
                this.setState({ characters: chars.results }, () => {
                    this.props.updateTotalPages(chars.totalPages);
                });
            });
    }

    render() {
        const characterList = this.state.characters.map(char => {
            return (
                <Character character={char} key={char.id}/>
            );
        });
        return (
            <Fragment>
                <ul className={styles.Characters}>
                    {characterList}
                </ul>
            </Fragment>
        );
    }
}

