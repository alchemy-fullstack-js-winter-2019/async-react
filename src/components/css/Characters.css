ul {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
}

li {
  list-style-type: none;
  position: relative;
  width: 50%;
}

h2 {
  font-size: 1.1em;
  width: 250px;
}