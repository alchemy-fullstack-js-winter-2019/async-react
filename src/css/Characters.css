.characters {
  font-family: sans-serif;
  ul {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    color: white;
    
    li {
      padding: 20px;
      background: indigo;
      list-style: none;
      border: 5px black;
      width: 200px;
      margin: 10px;
    }
  }
}
