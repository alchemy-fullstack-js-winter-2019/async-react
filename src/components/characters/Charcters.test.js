import React from 'react';
import renderer from 'react-test-renderer';
import Characters from './Characters';
jest.mock('../services/rickAndMortyApi.js');


describe('does things', () => {
    it('matches a snapchat', () => {
        const tree = renderer.create(
            <Characters/>
        );
        expect(tree).toMatchSnapshot();
    });
});