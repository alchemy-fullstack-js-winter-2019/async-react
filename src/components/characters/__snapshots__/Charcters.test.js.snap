// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does things matches a snapchat 1`] = `<ul />`;
